.github/workflows/windows-build-xpu.yml

```diff
@@ -0,0 +1,128 @@
+name: windows-build-xpu
+
+on:
+  push:
+    tags:
+      - ciflow/xpu/*
+  workflow_dispatch:
+
+      inputs:
+        win-runner:
+            required: false
+            type: string
+            default: "windows.4xlarge.nonephemeral"
+            description: |
+              Label of the runner this job should run on.
+        build-environment:
+            required: false
+            type: string
+            default: "xpu"
+            description: Top-level label for what's being built/tested.
+        s3-bucket:
+            description: S3 bucket to download artifact
+            required: false
+            type: string
+            default: "gha-artifacts" 
+            
+env:
+    conda_env: 'xpu_venv'
+    conda_path: 'C:\Jenkins\Miniconda3'
+    python_version: '3.10'
+    USE_XPU: 1
+    VS2022INSTALLDIR: 'C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools'
+
+jobs:
+  Preparation:
+    runs-on: ["${{ inputs.win-runner }}"]
```

Comment:
Do we want to reuse existed windows runner? If so, could we reuse the workflow
https://github.com/pytorch/pytorch/blob/main/.github/workflows/_win-build.yml
and scripts
https://github.com/pytorch/pytorch/blob/main/.ci/pytorch/win-build.sh
directly with limited modification to add xpu support?

Reply:
Let's check the root cause of the MSVC 2019 build failure. And we also are able to check the existed windows runner whether has msvc 2022 installed or if it's possible to install to 2019 and 2022 in advance for those existed runner